api/utils/route.go

```diff
@@ -75,14 +76,16 @@ func (m *SSHRouteMatcher) RouteToServer(server RouteableServer) bool {
 		return m.routeToHostname(server.GetHostname())
 	}
 
-	ip, nodePort, err := net.SplitHostPort(server.GetAddr())
-	if err != nil {
-		return false
-	}
+	for _, addr := range []string{server.GetAddr(), server.GetPublicAddr()} {
+		ip, nodePort, err := net.SplitHostPort(addr)
+		if err != nil {
+			return false
+		}
```

Comment:
nit: I think this should be `continue`. Seems potentially confusing to have one malformed address prevent the other address from being dialable.